src/etc/htmldocck.py

```diff
@@ -146,53 +146,21 @@ def concat_multi_lines(f):
     if lastline is not None:
         print_err(lineno, line, "Trailing backslash at the end of the file")
 
-
-def get_known_directive_names():
-    def filter_line(line):
-        line = line.strip()
-        return line.startswith('"') and (line.endswith('",') or line.endswith('"'))
-
-    # Equivalent to `src/tools/compiletest/src/header.rs` constant of the same name.
-    with open(
-        os.path.join(
-            # We go back to `src`.
-            os.path.dirname(os.path.dirname(__file__)),
-            "tools/compiletest/src/directive-list.rs",
-        ),
-        "r",
-        encoding="utf8",
-    ) as fd:
-        content = fd.read()
-        return [
-            line.strip().replace('",', "").replace('"', "")
-            for line in content.split("\n")
-            if filter_line(line)
-        ]
-
-
-# To prevent duplicating the list of commmands between `compiletest` and `htmldocck`, we put
-# it into a common file which is included in rust code and parsed here.
-# FIXME: This setup is temporary until we figure out how to improve this situation.
-#        See <https://github.com/rust-lang/rust/issues/125813#issuecomment-2141953780>.
-KNOWN_DIRECTIVE_NAMES = get_known_directive_names()
-
 LINE_PATTERN = re.compile(
     r"""
     //@\s+
-    (?P<negated>!?)(?P<cmd>[A-Za-z0-9]+(?:-[A-Za-z0-9]+)*)
+    (?P<negated>!?)(?P<cmd>.+?)
+    [\s:]
```

Comment:
I basically had to extend the grammar of {Html,Json}DocCk directives to admit `:` as a legal separator besides whitespace.

Otherwise it would've either been included in the args if I hadn't relaxed `cmd`'s grammar (so `//@ has: ...` would've meant \[`has`, `: ...`]) which leads to bad diagnostics down the line like "unexpected argument" (because `: ...` would've been further split into \[`:`, `...`] by shlex) or it would've been accepted by compiletest but ~~rejected~~ ignored by HtmlDocCk (remember, I can't reject any unknown or malformed directive in HtmlDocCk since it no longer knows if it's a compiletest directive or not).